.ci/onnx/jenkins/runCI.sh

```diff
@@ -16,104 +16,287 @@
+            "--cleanup")
+                CLEANUP="true"
+                echo "[INFO] Cleanup will be performed"
+                ;;
+            "--rebuild")
+                REBUILD="true"
+                echo "[INFO] nGraph is going to be rebuilt"
+                ;;
+            "--ngraph-onnx-branch="*)
+                NGRAPH_ONNX_REPO_BRANCH="${i//${PATTERN}/}"
+                ;;
+            "--ngraph-onnx-sha="*)
+                NGRAPH_ONNX_REPO_SHA="${i//${PATTERN}/}"
+                echo "[INFO] Using nGraph-ONNX commit ${NGRAPH_ONNX_REPO_SHA}"
+                ;;
+            "--backends="*)
+                BACKENDS="${i//${PATTERN}/}"
+                # Convert comma separated values into whitespace separated
+                BACKENDS="${BACKENDS//,/ }"
+                ;;
+            *)
+                echo "[ERROR] Unrecognized argument: ${i}"
+                printf ${HELP_MESSAGE}
+                exit -1
+                ;;
+        esac
+    done
 
-    docker exec "${DOCKER_CONTAINER}" bash -c 'rm -rf /root/build/* /root/ngraph_dist /root/python/dist'
+    echo "[INFO] Using nGraph-ONNX branch ${NGRAPH_ONNX_REPO_BRANCH}"
+    echo "[INFO] Backends tested: ${BACKENDS}"
+
+    return 0
 }
 
-# Function cleanup() removes items created during script execution
 function cleanup() {
-    set -x
+    # Performs cleanup of artifacts and containers from previous runs.
+    local container_name_pattern="${DOCKER_CONTAINER_NAME_PATTERN/<OPERATING_SYSTEM>/*}"
+    echo "[INFO] Performing cleanup"
+    docker rm -f "$(docker ps -a --format="{{.ID}}" --filter="name=${container_name_pattern}")"
+    rm -rf "${NGRAPH_ROOT_ABS_PATH}/python/dist/ngraph*.whl"
+    rm -rf "${NGRAPH_ROOT_ABS_PATH}/python/*.so"
+    rm -rf "${NGRAPH_ROOT_ABS_PATH}/python/build"
+    rm -rf "${NGRAPH_ONNX_REPO_PATH}"
+
+    return 0
+}
+
+function check_ngraph_onnx_repo() {
+    # Verifies if nGraph-ONNX repository is present
+    local ngraph_onnx_git="${NGRAPH_ONNX_REPO_PATH}/.git"
+    if [ -d "${ngraph_onnx_git}" ]; then
+        # 0 - true
+        return 0
+    else
+        # 1 - false
+        return 1
+    fi
+}
+
+function ngraph_onnx_rev_parse() {
+    # Returns the result of git rev-parse on nGraph-ONNX repository.
+    local rev_parse_args="${1}"
+    local previous_dir="$(pwd)"
+    local ngraph_onnx_dir="${NGRAPH_ONNX_REPO_PATH}"
+    cd "${ngraph_onnx_dir}"
+    local result="$(git rev-parse ${rev_parse_args} HEAD)"
+    cd "${previous_dir}"
+    echo "${result}"
+
+    return 0
+}
+
+function checkout_ngraph_onnx_repo() {
+    # Switches nGraph-ONNX repository to commit SHA
+    local rev="${1}"
+    local previous_dir="$(pwd)"
+    cd "${NGRAPH_ONNX_REPO_PATH}"
+    git checkout "${rev}"
+    cd "${previous_dir}"
+
+    return 0
+}
+
+function run_ci() {
+    # Builds necessary Docker images and executes CI
+    for dockerfile in $(find ${WORKSPACE}/${NGRAPH_ONNX_CI_PATH}/dockerfiles -maxdepth 1 -name *.dockerfile -printf "%f"); do
```

Comment:
The `-printf` option is not in `POSIX` `find`. This line will not work on a Mac.

Reply:
fixed